File: Modul6/Bahasa_Python/PRAK605-2310817210012-Nazmi-Hakim.py
def perkalian(matrixa, matrixb):
    result = [[0 for _ in range(len(matrixa))] for _ in range(len(matrixa))]

    for i in range(len(matrixa)):
        for j in range(len(matrixa)):
            for k in range(len(matrixa)):
                result[i][j] += matrixa[i][k] * matrixb[k][j]

    return result

def inputmatrix(n):
    matrix = []
    for _ in range(n):
        baris = list(map(int, input().split()))
        matrix.append(baris)
    return matrix

def cetakmatrix(matrix):
    for baris in matrix:
        print(" ".join(map(str, baris)))

n = int(input())
print("Matriks A")
matrixa = inputmatrix(n)
print("Matriks B")
matrixb = inputmatrix(n)
hasilmatrix = perkalian(matrixa, matrixb)
print("Matriks AXB")
cetakmatrix(hasilmatrix)

File: Modul6/Bahasa_C/PRAK605-2310817210012-Nazmi-Hakim.c
#include <stdio.h>

void perkalianmatrix(int n, int matrixa[n][n], int matrixb[n][n], int hasil[n][n]) {
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            hasil[i][j] = 0;
            for (int k = 0; k < n; k++) {
                hasil[i][j] += matrixa[i][k] * matrixb[k][j];
            }
        }
    }
}

void inputmatrix(int n, int matrix[n][n]) {
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            scanf("%d", &matrix[i][j]);
        }
    }
}

void cetakmatrix(int n, int matrix[n][n]) {
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            printf("%d ", matrix[i][j]);
        }
        printf("\n");
    }
}

int main() {
    int n;
    scanf("%d", &n);
    int matrixa[n][n], matrixb[n][n], hasilmatrix[n][n];
    printf("Matrix A:\n");
    inputmatrix(n, matrixa);
    printf("Matrix B:\n");
    inputmatrix(n, matrixb);
    perkalianmatrix(n, matrixa, matrixb, hasilmatrix);
    printf("Matrix AxB:\n");
    cetakmatrix(n, hasilmatrix);
    return 0;
}